modules/tensor_mechanics/src/materials/MaterialRankTwoDirectionScalar.C

```diff
@@ -0,0 +1,63 @@
+//* This file is part of the MOOSE framework
```

Comment:
It doesn't look like MaterialRankTwoDirectionScalar and MaterialRankTwoScalar are different in any substantial way.  What I had in mind was to have a set of models, each which only exposed the parameters applicable to it. High level things to change:
1) Drop 'Material' from the beginning of the name of all of these
2) If you just want to mimic what we currently do for the AuxKernels, you could have just two new classes: `RankTwoTensorScalar` and `RankTwoTensorComponent`. You could accomplish that by renaming `MaterialRankTwoTensorComponent` -> `RankTwoTensorComponent` and `MaterialRankTwoDirectionScalar`->`RankTwoTensorScalar` and deleting `MaterialRankTwoScalar`.
3) You could stop at 2) and have something like what we currently have, but we can do a little better than that by splitting the (renamed) `RankTwoTensorScalar` into a few separate classes that only have required parameters:
a) A class that computes the scalar reductions that don't require `point1`, `point2`, or `direction`. Don't provide those as parameters. You could use `RankTwoTensorScalar` as the name for this one. 
b) A class that computes the stress in a specified direction, and requires `direction`. You could call it `RankTwoScalarDirectionalComponent`
c) A class that computes the radial, hoop, or axial stress and requires `point1` and `point2`. You could call it `RankTwoCylindricalComponent`